packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -202,39 +203,23 @@ function transform(codeGen: CodeGen): t.Expression {
         return res;
     }
 
-    function transformIfBlock(ifBlockNode: IfBlock): t.Expression | t.Expression[] {
-        const expression = transformChildren(ifBlockNode);
-        let res: t.Expression | t.Expression[];
-
-        if (t.isArrayExpression(expression)) {
-            // Bind the expression once for all the template children
-            const testExpression = codeGen.bindExpression(ifBlockNode.condition);
+    function transformIfBlock(ifBlock: IfBlock | ElseifBlock): t.Expression {
```

Comment:
`conditionalParentBlock` might be a better name since it needs to encompass both `if` and `elseif`. It's what's currently used in a check I added on `ast.ts`. Open to other ideas too, just not `ifBlock` since that's inaccurate and misleading.

This applies for both the function name and the parameter.